### ISSUE TITLE
Add support for Debian binary package (deb/ipk)

### ISSUE THREAD

**Issue by jetm** [ouch-org/ouch#419]
Hi,

Thank you for creating `ouch`. It's a great tool. Save time remembering all the flags for every different compress file.

Maybe `ouch` was not designed for binary packages, like Debian. It includes several compressed files (`control.tar.gz`, `data.tar.xz`) inside and can get them using `ar` command. `file` gives the type file, in this case it reports: 
```
Debian binary package (format 2.0), with control.tar.gz, data compression xz
```

**Comment by marcospb19** [comment]
Thanks for the issue, glad to see you like Ouch :) 

I can't think of any problems with supporting `.deb`.

Yeah it would be the first of the kind (binary file with specific files in), but I believe it wouldn't clash with our current usage.

If you can do:
```
ouch compress a b c d pkg.deb
# and
ouch decompress pkg.deb
```

With no required extra args, then it's OK for it to be supported :)

If we need other specific args, we'd need to create a new subcommand, like `ouch bundle deb ...`, and in that case, I'd start questioning if that's really easier/more intuitive than just using the usual tools.

**Comment by jetm** [comment]
I was thinking more about the `decompress` command, not the `compress` because Debian has its own way to do so. 

It would be just extending `decompress` functionality based on the file type. Something like this,
```
ouch decompress pkg.deb
Detecting Debian binary package
decompressing pkg.deb
decompressing control tarball
decompressing data tarball
...
```

And that's it. It would be easier doing like that because no need to take care of the compression.

**Comment by marcospb19** [comment]
Makes sense! I never had to compress a `.deb` archive, but countless times I had to decompress it. It's certainly the most common use-case.

This would be the first format in Ouch that doesn't get compressed, similar to .rar if we end up adding it, @figsoda how do you feel about this one?



Unless

**Comment by figsoda** [comment]
I don't know the details of the `deb` format, but if it's like `zip`, I think we can support compression with a (arbitrary?) compression method that can be later extended to be customizable.

I actually opened an issue a while ago for the `ar` format and also lumped in `deb`: #201

**Comment by marcospb19** [comment]
> I think we can support compression with a (arbitrary?) compression method

We can choose a default compression method, the problem is providing more required fields that can't have a default and can't be omitted, I didn't even google it, but I'm assuming that's the case here, do you know @jetm?

**Comment by jetm** [comment]
I wouldn't bother with the compression method, just the decompression. Like you said it, the most common case for Debian package is decompress.
